### PR TITLE
Add mtenn package

### PR DESCRIPTION
**Pull request by hmacdope**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [X] Source is from official source.
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [X] If static libraries are linked in, the license of the static library is packaged.
- [X] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [X] Build number is 0.
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [X] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by hmacdope** [comment]
@mikemhenry @kaminow would you be maintainers on this package?

**Comment by kaminow** [comment]
definitely! I'll need some guidance at first though since I'm not really sure what all it entails

**Comment by hmacdope** [comment]
I would like this package to remain `noarch` with failing windows build, caused by missing deps.

**Comment by mikemhenry** [comment]
I am willing to maintain this package

**Comment by hmacdope** [comment]
@conda-forge/help-python this should be ready to roll, was going to leave as noarch with no windows builds due to missing deps, but let me know if this is the wrong approach.

File: recipes/mtenn/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+{% set version = "0.4.0" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+
+  - url: https://github.com/choderalab/{{ name }}/archive/{{ version }}.tar.gz
+    sha256: b78f6cf8ec9331c6e0772d55dfc7858cb61fd7da562a7bb47423811feddf2db4
```
Comment (ocefpaf) [inline recipes/mtenn/meta.yaml:11]:
```suggestion
source:
  url: https://github.com/choderalab/{{ name }}/archive/{{ version }}.tar.gz
  sha256: b78f6cf8ec9331c6e0772d55dfc7858cb61fd7da562a7bb47423811feddf2db4
```

File: recipes/mtenn/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
```
Comment (ocefpaf) [inline recipes/mtenn/meta.yaml:15]:
```suggestion
  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
```

**Pull request merged**